ui/src/components/Publication/Creation/MainText.tsx

```diff
@@ -143,6 +145,57 @@ const MainText: React.FC = (): React.ReactElement | null => {
         updateReferences
     } = Stores.usePublicationCreationStore();
 
+    const [editReferenceModalVisible, setEditReferenceModalVisible] = React.useState(false);
+    const [modalReferenceInfo, setModalReferenceInfo] = React.useState<Interfaces.Reference>({
```

Comment:
Showing the modal and editing the selected reference can be achieved with a single state.
The more states you add, the harder will be to test this component 😄 

You can have something like:
const [selectedReference, setSelectedReference] = useState(null); // null by default

and then for the Modal to show/hide just Boolean(selectedReference)